### PR TITLE
Implement GUI system, Part 1

### PR DESCRIPTION
**Pull request by utilForever**
This revision includes:
- Implement GUI system (#8)
  - Add library 'gl3w' to use OpenGL
  - Create structs
    - Inputs: This struct stores the position of mouse and the state of key/button
    - Size: This struct stores the size of the component for mouse picking
    - Color: This struct stores the color of the component
    - TextRenderer: This struct stores the data for rendering text at position with color
    - ShapeRenderer: This struct stores the functor for rendering shape
    - VertexPC: This struct stores the data for the position and color
    - CashButtonToggler: This struct stores the value of threshold to toggle the owner button depending on the cash threshold held by the player
    - Button: This struct defines basic properties of a button
    - Hoverable: This struct stores the value of normal/hover color
    - Resources: This struct stores resource handles
  - Add functions
    - OnMouseButtonPressed(): Callback when the mouse button is pressed
    - OnMouseButtonReleased(): Callback when the mouse button is released
    - CreateTowerButton(): Create a tower button
    - DrawBox(): Draws an box shape
    - UpdateShapeRenderSystem(): Draws entities that contains the component 'ShapeRenderers'
    - DrawLines(): Draw lines for Position + Color rendering
    - Update(): Updates game system according to delta time
    - Render(): Renders game system
    - PrepareForPC(): Prepare the rendering pipeline for Position + Color rendering
    - BeginFrame(): Start a rendering frame. This will clear background to black and set defaults render states
    - CreateProgram(): Create a shader program from vertex/fragment shader source
    - CreateVertexBuffer(): Create an empty vertex buffer
  - Refactor code
    - Change the type 'double' to 'float'
    - Remove unnecessary included headers
    - Move files 'Game.hpp/cpp' to parent folder
    - Delete file 'GameTests.cpp'
  - Update version of the library 'SFML' to 'latest' due to Visual Studio 2022 support
  - Add 'Game.hpp/cpp' to ingore list in Codecov

### CODE REVIEW

**Review by red1108 — APPROVED** [review]
Drowbox에서 왜 점을 8개를 만드나요?

**Comment by utilForever** [comment]
@red1108 Box는 3차원이니까 8개의 점이 필요합니다. 다만 우리는 z축을 사용하지 않기 때문에 x/y 값만 대입할 뿐입니다.

File: Includes/CubbyTower/Components/Hoverable.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#define CUBBYTOWER_HOVERABLE_HPP
+
+namespace CubbyTower
+{
+//!
+//! \brief Hoverable struct.
+//!
+//! This struct stores the value of normal/hover color.
+//!
+struct Hoverable
```
Comment (IHIHHIHI) [inline Includes/CubbyTower/Components/Hoverable.hpp:17]: 이건 마우스로 집어서 옮길때 색이 달라지는 물체를 나타내는건가요?

Comment (utilForever) [reply Includes/CubbyTower/Components/Hoverable.hpp:17]: 버튼 위에 마우스 커서를 갖다댈 때 색이 달라질 수 있음을 나타냅니다.

File: Sources/CubbyTower/Helpers/ShapeHelpers.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+#include <CubbyTower/Helpers/RenderingHelpers.hpp>
+#include <CubbyTower/Helpers/ShapeHelpers.hpp>
+
+namespace CubbyTower::Shape
+{
+int DrawBox(VertexPC* vertices, const Position& position, float width,
+            float height, const Color& color)
+{
+    vertices[0].position.x = position.x - width;
```
Comment (IHIHHIHI) [inline Sources/CubbyTower/Helpers/ShapeHelpers.cpp:15]: 이러면 width * 2 , height * 2 의 박스가 생기는거 맞나요??

Comment (utilForever) [reply Sources/CubbyTower/Helpers/ShapeHelpers.cpp:15]: 맞습니다.

**Review by IHIHHIHI — APPROVED** [review]

**Pull request merged**